pandas/core/indexing.py

```diff
@@ -443,6 +443,35 @@ def loc(self) -> _LocIndexer:
         8          4       5
         9          7       8
 
+        **Setting values using Series**
+
+        Assign column using Series
+
+        >>> df.loc[:, 'shield'] = pd.Series({7: 8, 8: 10, 9: 13})
+        >>> df
+           max_speed  shield
+        7          1       8
+        8          4      10
+        9          7      13
+
+        Assign partial column with labels in dataframe missing from Series
```

Comment:
Could you add something like the Series is aligned to the Index of the DataFrame

Reply:
Does what I added now in the latest commit look like what you were asking for? If not, can you clarify what to put a bit more?

Reply:
I thought about something like Assigning a Series with non matching Index aligns the right hand side to the Index of the left.